File: JS/数组/长度最小的子数组（滑动窗口）/209_长度最小的子数组.js
var minSubArrayLen = function(target, nums) {
    let left = 0,ans = Infinity
    let sum = 0
    for(let i=0; i<nums.length; i++){
        sum+=nums[i]
        if(sum >= target){
            while(sum - nums[left] >= target){
                sum -= nums[left]
                left ++
            }
            ans = Math.min(i-left+1, ans)
        }
    }
    return ans === Infinity ? 0 : ans
};

nums = [2,3,1,2,4,3]
console.log(minSubArrayLen(7, nums))

File: Python/面试经典150题/滑动窗口/209_长度最小的子数组.py
from typing import List
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        n = len(nums)
        left, right = 0, 0
        ans = n + 1
        sum = 0
        while right < n:
            sum += nums[right]
            while sum >= target:
                ans = min(ans, right - left + 1)
                sum -= nums[left]
                left += 1
            right += 1
        return ans if ans != n + 1 else 0

target = 7
nums = [2,3,1,2,4,3]
print(Solution().minSubArrayLen(target, nums))